src/lib/node/incoming_message.ts

```diff
@@ -12,4 +12,36 @@ export interface IncomingCommandNodeSetValue extends IncomingCommandNodeBase {
   value: unknown;
 }
 
-export type IncomingMessageNode = IncomingCommandNodeSetValue;
+export interface IncomingCommandNodeRefreshInfo
+  extends IncomingCommandNodeBase {
+  command: NodeCommand.refreshInfo;
+}
+
+export interface IncomingCommandNodeGetDefinedValueIDs
+  extends IncomingCommandNodeBase {
+  command: NodeCommand.getDefinedValueIDs;
+}
+
+export interface IncomingCommandNodeGetValueMetadata
+  extends IncomingCommandNodeBase {
+  command: NodeCommand.getValueMetadata;
+  valueId: ValueID;
+}
+export interface IncomingCommandNodeAbortFirmwareUpdate
+  extends IncomingCommandNodeBase {
+  command: NodeCommand.abortFirmwareUpdate;
+}
+export interface IncomingCommandNodeBeginFirmwareUpdate
+  extends IncomingCommandNodeBase {
+  command: NodeCommand.beginFirmwareUpdate;
+  buffer: Buffer;
```

Comment:
Our message format is limited to what can be represented in JSON.

Reply:
Sending files as b64 encoded strings over web socket seems a bit weird too. How large are these generally ?

Reply:
Right, while Buffer can be represented in JSON in JS, I have no idea about Python. Regarding the size I guess we need @marcelveldt advice ;)

Reply:
Let's leave this command out for now so we can move forward with this PR. We can then do a separate discussion about this. 

btw, if you have some extra time, could use some help with reviewing my PRs :)

Reply:
firmware update is pretty advanced so I'd say leave it out for now. Users can fire up zwave2mqtt control panel to do firmware updates so there's a way.
If it needs to be added some day, b64 encoded is very common to send binary data in a json repsonse.

Reply:
The problem with sending big payloads is that your web socket connection will not be used for anything else while the big file is being send. So any event will be lost .